Week1/homework/app.js

```diff
@@ -1,11 +1,169 @@
+      language: 'English',
+      author: 'By George R. R. Martin',
+    },
+    'harry-potter-and-the-goblet-of-fire': {
+      title: 'Harry Potter and the Goblet of Fire',
+      cover:
+        'https://upload.wikimedia.org/wikipedia/en/c/c7/Harry_Potter_and_the_Goblet_of_Fire.jpg',
+      language: 'English',
+      author: 'By J. K. Rowling',
+    },
+    'the-book-thief': {
+      title: 'The Book Thief',
+      cover:
+        'https://upload.wikimedia.org/wikipedia/en/8/8f/The_Book_Thief_by_Markus_Zusak_book_cover.jpg',
+      language: 'English',
+      author: 'By Markus Zusak',
+    },
+    'angels-and-demons': {
+      title: 'Angels and Demons',
+      cover: 'https://upload.wikimedia.org/wikipedia/en/5/5f/AngelsAndDemons.jpg',
+      language: 'English',
+      author: 'By Dan Brown',
+    },
+    'harry-potter-and-the-order-of-the-phoenix': {
+      title: 'Harry Potter and the Order of the Phoenix',
+      cover:
+        'https://upload.wikimedia.org/wikipedia/en/7/70/Harry_Potter_and_the_Order_of_the_Phoenix.jpg',
+      language: 'English',
+      author: 'By J. K. Rowling',
+    },
+  };
+
+  // I created another function, which takes books object as a parameter  Object.keys() method returns an array of a given object's own property names
+  // in the same order as we get with a normal loop. After getting array of object names, I used forEach method which is
+  // an array method to iterate through the names in the array.
+  // Inside forEach method, I used The Document method getElementById()
+  // that returns an Element object representing the element whose id property matches the specified string.
+  // Then created div element by using document.createElement() method, and used setAttribute method to set class attribute and its value on div.
+  // Here I wanted to be DRY, I created a re-useable function to create and append and set attribute.
+  // Here I used element[property] bracket notation, since I'm using the string passed to the property parameter.
+  // if you use dot notation JS thinks that I'm looking for a property that its name is property.
+  // Used if statement to set alt attribute only when the elementName is img.
+  // For how to use bracket notation:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors
+
+  const listOfBooks = obj =>
+    Object.keys(obj).forEach(item => {
+      const booksContainer = document.getElementById('books-container');
+
+      const bookDiv = document.createElement('div');
+      bookDiv.setAttribute('class', 'book-info');
+
+      const createElementAndSetAttr = (elementName, name, value, altName, altValue) => {
+        const element = document.createElement(elementName);
+        element.setAttribute(name, value);
+        if (elementName === 'img') {
+          element.setAttribute(altName, altValue);
+        }
+        bookDiv.appendChild(element);
```

Comment:
If you note closely, the function name doesn't correctly represent the actions inside it. For example, you are creating an element and attaching an attribute to it but you are also attaching it to the container. I would suggest that you **return** the new element from this function and append it to `bookDiv` where it's being created. Something like:

```
bookDiv.appendChild(createElementAndSetAttr('img', 'src', obj[item].cover, 'alt', obj[item].title));
```

or

```
const bookImage = createElementAndSetAttr('img', 'src', obj[item].cover, 'alt', obj[item].title);
bookDiv.appendChild(bookImage);
```

Same goes for the `createElementAndAppend` function 🙂

Reply:
I'm not sure if I got you well! I created another function and I append child to it! Please have a look if this what you've expected!